db/dbformat.h

```diff
@@ -808,6 +829,21 @@ class IterKey {
   size_t buf_size_;
   char space_[39];  // Avoid allocation for short keys
   bool is_user_key_;
+  // Below variables are only used by user-defined timestamps in MemTable only
+  // feature for iterating keys in an index block or a data block.
+  //
+  // We will alternate between buf_ and secondary_buf_ to hold the key. key_
+  // will be modified in accordance to point to the right one. This is to avoid
+  // an extra copy when we need to copy some shared bytes from previous key
+  // (delta encoding), and we need to pad a min timestamp at the right location.
+  char* secondary_buf_;
+  char space_for_secondary_buf_[39];  // Avoid allocation for short keys
```

Comment:
This probably wouldn't cause any issues in practice but since `secondary_buf_` can potentially point to `space_for_secondary_buf_`, it would be nice to have these two ordered the other way around. (Technically, `secondary_buf_` currently gets constructed before and destroyed after `space_for_secondary_buf_`.) Also, we could introduce a named constant for the size of the inline buffers (39).

Reply:
That's a good point, thank you for the suggestion!